### ISSUE TITLE
BelongsToJson returns wrong results

### ISSUE THREAD

**Issue by regi-summatech** [staudenmeir/eloquent-json-relations#103]
Hi, I stumbled upon this package while looking for something that I can use to implement a feature in a project. The idea here is I have a questionnaire in which an answer to a question has follow-up questions for that answer.

Here is my questions table:
![1a_questions_table](https://github.com/staudenmeir/eloquent-json-relations/assets/143600158/c7bc8dc3-dabd-4e93-8818-db05b0baf560)

Here is my subquestions table:
![1b_subquestions_table](https://github.com/staudenmeir/eloquent-json-relations/assets/143600158/69956616-da5e-4922-b906-f00dae397e6d)

What I wanted to happen here based on the tables shown in the screenshots above is to create a relation that there are many questions for an answer. Here is my Questions model:
![2a_questions_model](https://github.com/staudenmeir/eloquent-json-relations/assets/143600158/92f0a20e-5a58-4be3-89a6-71149023aa35)

I was using hasManyJson before but I was not getting any results until I saw #61 and replaced it with belongsToJson instead. I was able to get results while testing in Postman. However, I am getting the wrong results:
![image](https://github.com/staudenmeir/eloquent-json-relations/assets/143600158/97965858-eb56-423c-a407-4b8e40084d96)

Based on the tables above, I should be able to get all 5 of them since all of the values in answer_id in the subquestions table are equal to the answer_id 1 in the options column. My theory is that the answer_id in belongsToJson is looking at the id column. I tried defining answer_id as the foreign key value in the belongsToJson method:
![image](https://github.com/staudenmeir/eloquent-json-relations/assets/143600158/1c4d4c05-abcc-4eeb-8d83-1a3c2ba9bb3b)

However, when I do this instead I am not getting any subquestions related to the answers. The question here is if the use case I'm currently doing is supported by the package and if it is, does that mean I am doing something wrong?

**Comment by staudenmeir** [comment]
Hi @regi-summatech,
>  I tried defining answer_id as the foreign key value in the belongsToJson method:

Swap the second and third argument (`answer_id` is a custom owner key in this case):

```php
return $this->belongsToJson(KYCFollowupQuestion::class, 'options[]->answer_id', 'answer_id');
```

**Comment by regi-summatech** [comment]
Hi, I tried what you suggested and I can get all the questions when I provide the specific value in the model (`FollowupQuestion::find(n)->followup_questions`). I haven't added how I was trying to access it before but to give context, I am using `Question::with('followup_questions')->get()` which might be wrong because I was only getting the last record. Thank you for your time. Closing this now.
